projects/js-packages/components/components/product-detail/index.jsx

```diff
@@ -0,0 +1,191 @@
+
+/**
+ * Internal dependencies
+ */
+import Text, { H3 } from '../text/index.jsx';
+import { getIconBySlug, CheckmarkIcon } from '../product-icons/index.jsx';
+import ProductPrice from '../product-price/index.jsx';
+import styles from './style.module.scss';
+import Button from '../button/index.jsx';
+
+/**
+ * Product Detail Card Header component.
+ *
+ * @param {object} props       - Component props.
+ * @param {string} props.title - Product title.
+ * @returns {React.Component}  ProductDetailHeader react component.
+ */
+function ProductDetailHeader( { title = __( 'Popular upgrade', 'jetpack' ) } ) {
+	return (
+		<div className={ styles[ 'card-header' ] }>
+			<Icon icon={ star } className={ styles[ 'product-bundle-icon' ] } size={ 24 } />
+			<Text variant="label">{ title }</Text>
+		</div>
+	);
+}
+
+/**
+ * Icons composition for a bundle product,
+ * based on the list of supported products.
+ *
+ * @param {object} props         - Component props.
+ * @param {Array} props.products - List of supported products.
+ * @returns {React.Component}    Bundle product icons react component.
+ */
+function ProductIcons( { products } ) {
+	return (
+		<div className={ styles[ 'product-bundle-icons' ] }>
+			{ products.map( ( product, index ) => {
+				const ProductIcon = getIconBySlug( product );
+				const ProIcon = ProductIcon ? ProductIcon : () => null;
+
+				return (
+					<Fragment key={ index }>
+						<ProIcon size={ 24 } />
+
+						{ index !== products.length - 1 && (
+							<Icon
+								className={ styles[ 'plus-icon' ] }
+								key={ `icon-plugs${ index * 2 + 1 }` }
+								icon={ plus }
+								size={ 16 }
+							/>
+						) }
+					</Fragment>
+				);
+			} ) }
+		</div>
+	);
+}
+
+/**
+ * Product Detail component.
+ *
+ * @param {object} props                  - Component props.
+ * @param {string} props.slug             - Product slug.
+ * @param {string} props.title 			  - Product title.
+ * @param {string} props.description      - Product description.
+ * @param {Array}  props.features         - Features list of the product.
+ * @param {boolean} props.isCard          - Add the styles to look like a card.
+ * @param {boolean} props.isBundle        - Whether or not the product is a bundle.
+ * @param {Array} props.supportedProducts - List of supported products (for bundles).
+ * @param {Object} props.pricing 	      - Product Pricing object.
+ * @param {boolean} props.hasRequiredPlan - Whether or not the product has the required plan.
+ * @param {boolean} props.isLoading       - Applies the isLoading style to the component.
+ * @param {string} props.className        - A className to be concat with default ones.
+ * @param {Function} props.onAdd          - Callback function to be executed on click on Add button.
+ * @param {string} props.addProductUrl    - The checkout URL to add/buy the product.
+ * @returns {React.Component}               ProductDetail react component. Optional.
+ */
+const ProductDetail = ( {
```

Comment:
Now that we made it generic, this name looks strange since I don't think the price is part of the product detail. Not a big deal, but just sharing my concern.

It's looks more `ProductOffer` to me

Reply:
The price is part of the product, among other details. 

> It's looks more ProductOffer to me

yes, we can rename it.

Reply:
https://github.com/Automattic/jetpack/pull/23842/commits/52cd1e972a10fcbf3ed50bc330bc7e8c6f1f3093